libs/blocks/tabs/tabs.js

```diff
@@ -267,36 +276,41 @@ const init = (block) => {
   const tabContent = createTag('div', { class: 'tab-content' }, tabContentContainer);
   block.append(tabContent);
 
+  const isRadio = block.classList.contains('radio');
   // Tab List
   const tabList = rows[0];
   tabList.classList.add('tabList');
-  tabList.setAttribute('role', 'tablist');
+  tabList.setAttribute('role', isRadio ? 'radiogroup' : 'tablist');
   const tabListContainer = tabList.querySelector(':scope > div');
   tabListContainer.classList.add('tab-list-container');
+  const tabListLabel = config.pretext;
+  if (tabListLabel)tabList.setAttribute('aria-label', tabListLabel);
+
   const tabListItems = rows[0].querySelectorAll(':scope li');
   if (tabListItems) {
     const pillVariant = [...block.classList].find((variant) => variant.includes('pill'));
     const btnClass = pillVariant ? handlePillSize(pillVariant) : 'heading-xs';
     tabListItems.forEach((item, i) => {
       const tabName = config.id ? i + 1 : getStringKeyName(item.textContent);
+      const controlId = `tab-panel-${tabId}-${tabName}`;
       const tabBtnAttributes = {
-        role: 'tab',
+        role: isRadio ? 'radio' : 'tab',
         class: btnClass,
         id: `tab-${tabId}-${tabName}`,
         tabindex: '0',
-        'aria-selected': (i === 0) ? 'true' : 'false',
-        'aria-controls': `tab-panel-${tabId}-${tabName}`,
+        [isRadio ? 'aria-checked' : 'aria-selected']: (i === 0) ? 'true' : 'false',
```

Comment:
Just a thought - it's not common throughout our codebase to use ternary operators when defining property names. These all work, I'm just hoping this pattern is clear for everyone that will make changes in the future.